File: app/src/org/koreader/launcher/JNILuaInterface.java
package org.koreader.launcher;

/* Declares methods that are exposed to lua via JNI
 *
 * Note: inspection reports unused methods but they are all used by lua functions.
 * See https://github.com/koreader/android-luajit-launcher/blob/master/assets/android.lua
 */
@SuppressWarnings("unused")
interface JNILuaInterface {

    void setClipboardText(final String text);
    void einkUpdate(int mode);
    void einkUpdate(int mode, long delay, int x, int y, int width, int height);
    void dictLookup(String text, String pkg, String action);
    void setFullscreen(boolean enabled);
    void setScreenBrightness(int brightness);
    void setScreenOffTimeout(int timeout);
    void setWakeLock(boolean enabled);
    void setWifiEnabled(boolean enabled);
    void showToast(String message);

    int download(String url, String name);
    int extractAssets();
    int getBatteryLevel();
    int getScreenOffTimeout();
    int getScreenAvailableHeight();
    int getScreenHeight();
    int getScreenWidth();
    int getStatusBarHeight();
    int hasClipboardTextIntResultWrapper();
    int hasExternalStoragePermission();
    int hasWriteSettingsPermission();
    int isCharging();
    int isDebuggable();
    int isEink();
    int isEinkFull();
    int isFullscreen();
    int isPackageEnabled(String pkg);
    int isWifiEnabled();
    int needsWakelocks();
    int openLink(String url);

    String getClipboardText();
    String getEinkPlatform();
    String getFlavor();
    String getName();
    String getNetworkInfo();
    String getProduct();
    String getVersion();
}

File: app/src/org/koreader/launcher/JNILuaInterface.kt
package org.koreader.launcher

/* Declares methods that are exposed to lua via JNI
 * See https://github.com/koreader/android-luajit-launcher/blob/master/assets/android.lua */

internal interface JNILuaInterface {
    fun canWriteSystemSettings(): Int
    fun dictLookup(text: String, action: String)
    fun dictLookup(text: String, pkg: String?, action: String)
    fun download(url: String, name: String): Int
    fun einkUpdate(mode: Int)
    fun einkUpdate(mode: Int, delay: Long, x: Int, y: Int, width: Int, height: Int)
    fun extractAssets(): Int
    fun getBatteryLevel(): Int
    fun getClipboardText(): String
    fun getEinkPlatform(): String
    fun getExternalPath(): String
    fun getFlavor(): String
    fun getName(): String
    fun getNetworkInfo(): String
    fun getProduct(): String
    fun getScreenBrightness(): Int
    fun getScreenOffTimeout(): Int
    fun getScreenAvailableHeight(): Int
    fun getScreenHeight(): Int
    fun getScreenWidth(): Int
    fun getStatusBarHeight(): Int
    fun getSystemTimeout(): Int
    fun getVersion(): String
    fun hasClipboardText(): Int
    fun hasExternalStoragePermission(): Int
    fun isCharging(): Int
    fun isDebuggable(): Int
    fun isEink(): Int
    fun isEinkFull(): Int
    fun isFullscreen(): Int
    fun isPackageEnabled(pkg: String): Int
    fun isWifiEnabled(): Int
    fun needsWakelocks(): Int
    fun openLink(url: String): Int
    fun performHapticFeedback(constant: Int)
    fun requestWriteSystemSettings()
    fun setFullscreen(enabled: Boolean)
    fun setClipboardText(text: String)
    fun setHapticOverride(enabled: Boolean)
    fun setScreenBrightness(brightness: Int)
    fun setScreenOffTimeout(timeout: Int)
    fun setWifiEnabled(enabled: Boolean)
    fun showToast(message: String)
    fun showToast(message: String, longTimeout: Boolean)
}
